File: templates/Uwp/Features/DeepLinking.MVVMLight/Activation/SchemeActivationData.cs
using System;
using System.Collections.Generic;
using System.Web;

namespace Param_ItemNamespace.Activation
{
    public class SchemeActivationData
    {
        private const string ProtocolName = "wtsapp";

        public string ViewModelName { get; private set; }

        public Uri Uri { get; private set; }

        public Dictionary<string, string> Parameters { get; private set; } = new Dictionary<string, string>();

        public bool IsValid => ViewModelName != null;

        public SchemeActivationData(Uri activationUri)
        {
            ViewModelName = SchemeActivationConfig.GetViewModelName(activationUri.AbsolutePath);

            if (!IsValid || string.IsNullOrEmpty(activationUri.Query))
            {
                return;
            }

            var uriQuery = HttpUtility.ParseQueryString(activationUri.Query);
            foreach (var paramKey in uriQuery.AllKeys)
            {
                Parameters.Add(paramKey, uriQuery.Get(paramKey));
            }
        }

        public SchemeActivationData(string viewModelName, Dictionary<string, string> parameters = null)
        {
            ViewModelName = viewModelName;
            Parameters = parameters;
            Uri = BuildUri();
        }

        private Uri BuildUri()
        {
            var viewModelKey = SchemeActivationConfig.GetViewModelKey(ViewModelName);
            var uriBuilder = new UriBuilder($"{ProtocolName}:{viewModelKey}");
            var query = HttpUtility.ParseQueryString(string.Empty);

            foreach (var parameter in Parameters)
            {
                query.Set(parameter.Key, parameter.Value);
            }

            uriBuilder.Query = query.ToString();
            return new Uri(uriBuilder.ToString());
        }
    }
}

File: templates/Uwp/Features/DeepLinking.MVVMLightVB/Activation/SchemeActivationData.vb
Imports System.Web

Namespace Activation
    Public Class SchemeActivationData
        Private Const ProtocolName As String = "wtsapp"

        Public Property ViewModelName As String

        Public Property Uri As Uri

        Public Property Parameters As Dictionary(Of String, String) = New Dictionary(Of String, String)()

        Public ReadOnly Property IsValid As Boolean
            Get
                Return ViewModelName IsNot Nothing
            End Get
        End Property

        Public Sub New(activationUri As Uri)
            ViewModelName = SchemeActivationConfig.GetViewModelName(activationUri.AbsolutePath)

            If Not IsValid OrElse String.IsNullOrEmpty(activationUri.Query) Then
                Return
            End If

            Dim uriQuery = HttpUtility.ParseQueryString(activationUri.Query)

            For Each paramKey In uriQuery.AllKeys
                Parameters.Add(paramKey, uriQuery.[Get](paramKey))
            Next
        End Sub

        Public Sub New(viewModelName As String, Optional parameters As Dictionary(Of String, String) = Nothing)
            Me.ViewModelName = viewModelName
            Me.Parameters = parameters
            Me.Uri = BuildUri()
        End Sub

        Private Function BuildUri() As Uri
            Dim viewModelKey = SchemeActivationConfig.GetViewModelKey(ViewModelName)
            Dim uriBuilder = New UriBuilder($"{ProtocolName}:{viewModelKey}")
            Dim query = HttpUtility.ParseQueryString(String.Empty)

            For Each parameter In Parameters
                query.[Set](parameter.Key, parameter.Value)
            Next

            uriBuilder.Query = query.ToString()
            Return New Uri(uriBuilder.ToString())
        End Function
    End Class
End Namespace